logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/DeleteMessageUseCase.kt

```diff
@@ -0,0 +1,57 @@
+package com.wire.kalium.logic.feature.message
+
+import com.benasher44.uuid.uuid4
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.conversation.ConversationId
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.suspending
+import com.wire.kalium.logic.sync.SyncManager
+import kotlinx.coroutines.flow.first
+import kotlinx.datetime.Clock
+
+class DeleteMessageUseCase(
+    private val messageRepository: MessageRepository,
+    private val userRepository: UserRepository,
+    private val clientRepository: ClientRepository,
+    private val syncManager: SyncManager,
+    private val messageSender: MessageSender
+) {
+
+    suspend operator fun invoke(conversationId: ConversationId, messageId: String, deleteForEveryOne: Boolean): Either<CoreFailure, Unit> {
+        syncManager.waitForSlowSyncToComplete()
+        val selfUser = userRepository.getSelfUser().first()
+
+        return suspending {
+            val generatedMessageUuid = uuid4().toString()
+
+            clientRepository.currentClientId().flatMap { currentClientId ->
+                val message = Message(
+                    id = generatedMessageUuid,
+                    content = if (deleteForEveryOne)
+                        MessageContent.DeleteMessage(messageId)
+                    else
+                        MessageContent.HideMessage(messageId, conversationId),
+                    conversationId = conversationId,
+                    date = Clock.System.now().toString(),
+                    senderUserId = selfUser.id,
+                    senderClientId = currentClientId,
+                    status = Message.Status.PENDING
+                )
+                messageSender.getRecipientsAndAttemptSend(conversationId, message)
```

Comment:
The `HideMessage` shouldn't be sent in the conversation of the message, because we only want to send the `HideMessage` to your own devices and therefore we should send it to the "self conversation". The "self conversation" is special conversation which is created by the backend and is of type `self`.

I I think you need to wait for https://github.com/wireapp/kalium/pull/280 to implement this though. If you don't want to wait for it add a TODO.